### DISCUSSION TITLE
Extend command with arguments when running  job

### DISCUSSION THREAD

**Discussion by VaZark** [aws/copilot-cli#5695] [Q&A] [answered]
The docs for [job run](https://aws.github.io/copilot-cli/docs/commands/job-run/) does not specify if there is a way to add arguments to extend the default docker command during a manual invocation. 

Is there away to extend the `command` specified in the manifest with arguments?

**Comment by Lou1415926** [comment]
No unfortunately. `job run` is designed to invoke the deployed job once, under the hood it is a ECS task that is built respective to an ECS task definition. Changing the job configuration like `command`, requires a redeployment of the job so that the ECS task definition is updated.

You can see if `copilot task run --generate-cmd app/env/job` work for you - after running the command, modify the output so that you have the `--command` value you want.

**Reply by VaZark** [reply]
That's a shame. The primary reason for reaching for manual task runs is one-off runs with custom arguments. I'll generate a manual job for the moment

**Reply by VaZark** [reply]
Is there a way to refer to an existing ECR repo associated with another copilot svc in a new manifest without hardcoding the repository URL with the account ID?

**Reply by Lou1415926** [reply, marked as answer]
> The primary reason for reaching for manual task runs is one-off runs

Sorry I thought by looking to use `job run`, you wanted to run the job with the command only once. Are you actually looking to run it regularly instead? In this case, creating another job with `on.schedule: 'none'` is the way to go. You will just run `copilot job run` every time you want to manually trigger it.

> Is there a way to refer to an existing ECR repo associated with another copilot svc in a new manifest without hardcoding the repository URL with the account ID?

There is no support for sharing images between two workloads unfortunately. I think your best bet in this case is indeed to write the image uri in `image.location` :(

**Reply by VaZark** [reply]
Thanks. That's what I ended up doing :)

I raised this question as in my current task, Django has a built-in feature to write and invoke CLI commands as a part of the framework in the format `python manage.py <custom_command>`. 

I was looking to create a manual job with `entrypoint= "python manage.py"` with a customisable `command` parameter so that I didn't have to set up a new job with all the associated infrastructure for each custom command. In my ideal case, I would've wanted to use something like  `copilot job run --name <existing_manual_task_name> --command <command string>`

I usually use the `svc exec` on an existing webserver but some commands are long-running and I didn't want to risk impacting performance.
